### PR TITLE
[DO NOT MERGE] Docs team edit for "Authenticating to the GitHub API" doc

### PR DESCRIPTION
**Pull request by lucascosti**
This is an edit of the "[Authenticating to the GitHub API](https://github.com/actions/actions-runner-controller/blob/master/docs/authenticating-to-the-github-api.md)" article by the GitHub docs team.

**Please don't merge this PR until the docs team is ready.**

### CODE REVIEW

**Comment by Link-** [comment]
@lucascosti convert it to a draft please and switch the status when you're ready for review.

**Comment by lucascosti** [comment]
👋 @Link-, this is ready for a review, but we'd like to hold off on merging until the docs team has also reviewed and is ready for the merge. 🙇‍♂️

File: docs/authenticating-to-the-github-api.md [hunk 1]
```diff
@@ -1,208 +1,238 @@
 # Authenticating to the GitHub API
 
-## Setting Up Authentication with GitHub API
+## Overview
 
-There are two ways for actions-runner-controller to authenticate with the GitHub API (only 1 can be configured at a time however):
+There are two ways for Actions Runner Controller (ARC) to authenticate with the GitHub API, although you must one use only one of the options at a time.
```
Comment (mumoshu) [inline docs/authenticating-to-the-github-api.md:5]:
```suggestion
There are two ways for Actions Runner Controller (ARC) to authenticate with the GitHub API, although you must use only one of the options at a time.
```

Nit but perhaps there's a redundant "use" here?

File: docs/authenticating-to-the-github-api.md [hunk 2]
```diff
@@ -1,208 +1,238 @@
...
+     kubectl create secret generic controller-manager \
+         -n actions-runner-system \
+         --from-literal=github_token=${GITHUB_TOKEN}
+     ```
+   * If you are using a Helm chart for your deployment, in your `values.yaml`, set `authSecret.github_token` with the value of the token.
 
+     For more information, see the [Helm chart README](https://github.com/actions/actions-runner-controller/blob/master/charts/actions-runner-controller/README.md).
 
-### Using without cert-manager
+## Deploying without using without cert-manager
```
Comment (mumoshu) [inline docs/authenticating-to-the-github-api.md:163]: Perhaps you meant to say:

```suggestion
## Deploying without using cert-manager
```

**Comment by lucascosti** [comment]
Thanks for the review @mumoshu, and for catching those typos.

I'm closing this PR in favor of the one opened in the internal GitHub docs repository for the same info.

**Pull request closed without merging**